src/store/mqtt/actions.ts

```diff
@@ -133,7 +135,7 @@ export function disconnect() {
   return async function(dispatch: Dispatch, getState: GetState) {
     dispatch({ type: constants.DISCONNECT_START });
     const state = getState();
-    if (!client || !state.mqtt.connected) {
+    if (!client && !client.connected) {
```

Comment:
this `&&` should be a `||` right?

Reply:
nope, javascript "short circuits" https://stackoverflow.com/questions/12554578/does-javascript-have-short-circuit-evaluation